aidial_assistant/json_stream/exceptions.py

```diff
@@ -1,10 +1,21 @@
 class JsonParsingException(Exception):
-    pass
+    def __init__(self, message: str, char_position: int):
+        super().__init__(
+            f"Failed to parse json string at position {char_position}: {message}"
+        )
 
 
 def unexpected_symbol_error(
     char: str, char_position: int
 ) -> JsonParsingException:
-    return JsonParsingException(
-        f"Failed to parse json string: unexpected symbol {char} at position {char_position}"
-    )
+    return JsonParsingException(f"Unexpected symbol {char}.", char_position)
```

Comment:
Maybe quote the symbol to make absolutely clear what it is.

`Unexpected symbol '{char}'.`